packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -207,15 +239,30 @@ const ChartMarkers = connect(({ modules, ui, client }) => ({
 
 class ChartTradeClass extends React.Component {
     state = {
-        active_category: null,
+        open_market: null,
     };
     bottomWidgets = ({ digits, tick }) => <ChartBottomWidgets digits={digits} tick={tick} />;
     topWidgets = ({ ...props }) => {
-        const { is_digits_widget_active, try_synthetic_indices } = this.props;
+        const { is_digits_widget_active, try_synthetic_indices, try_open_markets } = this.props;
+        if (try_synthetic_indices) {
+            this.setState({
+                open_market: {
+                    category: 'synthetic_index',
+                },
+            });
+        } else if (try_open_markets) {
+            const { category, subcategory } = this.props;
+            if (category) {
+                this.setState({
+                    open_market: { category, subcategory },
+                });
+            }
+        }
+
```

Comment:
just my opinion. as we passed `category`  from the upper component, why not apply this logic also there? this way, we only have one place which we decide about what market or category we want to get open